platforms/core-runtime/launcher/src/main/java/org/gradle/tooling/internal/provider/ProblemAdditionalDataRemapper.java

```diff
@@ -57,43 +66,39 @@ private void remapAdditionalData(Object message) {
             return;
         }
         InternalAdditionalData additionalData = ((DefaultProblemDetails) details).getAdditionalData();
-        if (!(additionalData instanceof InternalPayloadSerializedAdditionalData)) {
+        if (!(additionalData instanceof InternalPayloadSerializedAdditionalDataV2)) {
```

Comment:
Note: my assumption is that we are still in the provider context and we cannot get instances coming from different Gradle versions here. Please check that.

Reply:
We are on provider side, but here it still loads the `InternalPayloadSerializedAddtionalData` from the client.

But since we are on the provider side, it is safe to use the explicit class instead of the interface.

So I did that in my changes.